examples/step-32/step-32.cc

```diff
@@ -3727,11 +3727,11 @@ int main (int argc, char *argv[])
   using namespace Step32;
   using namespace dealii;
 
-  Utilities::MPI::MPI_InitFinalize mpi_initialization(argc, argv,
-                                                      numbers::invalid_unsigned_int);
-
   try
     {
+      Utilities::MPI::MPI_InitFinalize mpi_initialization(argc, argv,
+                                                          numbers::invalid_unsigned_int);
+
```

Comment:
Moving the code here is probably a bad idea. If an exception is thrown,  the MPI object on that process will be destroyed. This will call MPI Finalize which will dead lock because the other processes are not at the same place...